qp_klp/Step.py

```diff
@@ -326,23 +328,134 @@ def _generate_prep_file(self, config, input_file_path, seqpro_path,
 
         return gpf_job
 
-    def _generate_commands(self):
-        out_dir = self.pipeline.output_path
-        qclient = self.qclient
+    def _helper_process_fastp_report_dirs(self):
+        report_dirs = []
+
+        for root, dirs, files in walk(self.pipeline.output_path):
+            for dir_name in dirs:
+                if dir_name == 'fastp_reports_dir':
+                    # generate the full path for this directory before
+                    # truncating everything up to the QCJob directory.
+                    full_path = join(root, dir_name).split('QCJob/')
+                    report_dirs.append(join('QCJob', full_path[1]))
+
+        if report_dirs:
+            report_dirs.sort()
+            return 'tar zcvf reports-QCJob.tgz ' + ' '.join(report_dirs)
+        else:
+            # It is okay to return an empty list of commands if reports_dirs
+            # is empty. Some pipelines do not generate fastp reports.
+            return []
+
+    def _helper_process_blanks(self):
+        results = [x for x in listdir(self.pipeline.output_path) if
+                   x.endswith('_blanks.tsv')]
+
+        if len(results) > 0:
+            return 'tar zcvf sample-files.tgz' + ' ' + ' '.join(results)
+
+    def _helper_process_operations(self):
```

Comment:
I still don't understand the complexity of this method; it basically creates a list of commands that are abstracted via multiple variables and arrays; I mean, why not simply have:

```python
commands = [
    # convert logs
    'tar zcvf logs-ConvertJob-tgz ConvertJob/logs',
     ...
     ...
]

```

Reply:
This was what we ran with originally. Daniel suggested we generalize it:
https://github.com/qiita-spots/qp-knight-lab-processing/pull/67#discussion_r1223255298